### ISSUE TITLE
Grant permission on namespace successfully when disabling authorization

### ISSUE THREAD

**Issue by zymap** [apache/pulsar#7343]
**Describe the bug**

Using the master code, grant permission on namespace successfully when disabling authorization. I am not sure it is correct or not.

Using version 2.5.2 of  Pulsar, it will throw an error:

```
 bin/pulsar-admin namespaces grant-permission public/default --actions produce --role admin
Authorization is not enabled

Reason: HTTP 501 Not Implemented
```

**Comment by merlimat** [comment]
This was already fixed in 2.6, see #7074

**Comment by zymap** [comment]
Thanks
